samples/wifi/shell/README.rst

```diff
@@ -0,0 +1,107 @@
+
+.. contents::
+   :local:
+   :depth: 2
+
+This sample allows testing Nordic Wi-Fi chipsets.
+
+Requirements
+************
+
+The sample supports the following development kit:
+
+.. table-from-rows:: /includes/sample_board_rows.txt
+   :header: heading
+   :rows: nrf7002dk_nrf5340_cpuapp, nrf7002dk_nrf5340_cpuapp_ns, nrf7002dk_nrf5340_cpunet
+
+
+Overview
+********
+
+You can use this sample as a starting point to experiment and test Nordic Wi-Fi chipsets.
+
+The sample can be used to perform all Wi-Fi operations in 2.4/5/6GHz bands depending on the capabilities supported in the underlying
+chipset.
+
+You can associate and ping to any Wi-Fi capable access point in :abbr:`STA (Station)` mode.
+
+It can also support other Wi-Fi modes of operation depending on the underlying chipset capabilities.
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/wifi/shell`
+
+.. include:: /includes/build_and_run_ns.txt
+
+Currently, the following configurations are supported:
+
+* 7002 DK + QSPI
+* 7002 EK + SPIM
+
+You can build for the nRF7002 DK using the following command::
+
+   west build -b nrf7002dk_nrf5340_cpuapp
+
+You can build for the nRF7002 EK using the following command::
+
+   west build -b nrf5340dk_nrf5340_cpuapp -- -DSHIELD=nrf7002_ek
+
+
+Testing
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
+#. Scan for the Wi-FI networks. The output should be similar to this example::
+
+      uart:~$ wpa_cli scan
+      Scan requested
+      Num  | SSID                             (len) | Chan | RSSI | Sec
+      1    | dlink-7D64                       10    | 1    | 0    | WPA/WPA2
+      2    | abcd_527_24                      11    | 4    | 0    | Open
+      3    | ASUS_RTAX88U11g                  15    | 3    | 0    | WPA/WPA2
+      4    | TP-Link_6BA8                     12    | 9    | 0    | WPA/WPA2
+
+
+#. Connect to a preferred network::
+
+      uart:~$ wpa_cli connect ASUS_RTAX88U11g <passphrase>
+
+
+#. Wait for some time and check the connection status. fIf all works well, you should see something like this::
+
+      uart:~$ wpa_cli status
+      Reply: bssid=c0:06:c3:1d:cf:9e
+      freq=0
+      ssid=ASUS_RTAX88U11g
+      id=0
+      mode=station
+      pairwise_cipher=CCMP
+      group_cipher=CCMP
+      key_mgmt=WPA2-PSK
+      pmf=1
+      mgmt_group_cipher=BIP
+      wpa_state=COMPLETED
+      ip_address=192.168.0.206
+      address=f4:ce:36:00:00:16
+
+
+#. Initiate a ping and verify data connectivity::
+
+      uart:~$ net dns google.com
+      Query for 'google.com' sent.
+      dns: 142.250.74.46
+      dns: All results received
+
+      uart:~$ net ping 10 142.250.74.46
+      PING 142.250.74.46
+      28 bytes from 142.250.74.46 to 192.168.50.199: icmp_seq=0 ttl=113 time=191 ms
+      28 bytes from 142.250.74.46 to 192.168.50.199: icmp_seq=1 ttl=113 time=190 ms
+      28 bytes from 142.250.74.46 to 192.168.50.199: icmp_seq=2 ttl=113 time=190 ms
```

Comment:
Please list all the sample's dependencies at the bottom of the document, using the following syntax

Dependencies
************

This sample uses the following |NCS| libraries:

* :ref:``
* :ref:``

This sample uses the following `sdk-nrfxlib`_ libraries:

* :ref:``

This sample uses the following `sdk-zephyr`_ libraries:

* :ref:``

Reply:
You are right on that section being marked as optional. It was initially intended as optional because there might not necessarily be dependencies. However, if there are any, these should be listed. Probably this point should be clarified in the template.

Reply:
OK, will add.